src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_InvalidExpression.cs

```diff
@@ -0,0 +1,390 @@
+    }
+}
+";
+            string expectedOperationTree = @"
+IFieldInitializer (Field: System.Int32 Program.x) (OperationKind.FieldInitializerAtDeclaration, IsInvalid)
+  IConversionExpression (ConversionKind.Invalid, Implicit) (OperationKind.ConversionExpression, Type: System.Int32, IsInvalid)
+    IInvalidExpression (OperationKind.InvalidExpression, Type: Program, IsInvalid)
+      IOperation:  (OperationKind.None)
+";
+            VerifyOperationTreeForTest<EqualsValueClauseSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17598, "https://github.com/dotnet/roslyn/issues/17598")]
+        public void InvalidArrayInitializer()
+        {
+            string source = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var x = new int[2, 2] /*<bind>*/{ { { 1, 1 } }, { 2, 2 } }/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayInitializer (2 elements) (OperationKind.ArrayInitializer, IsInvalid)
+  IArrayInitializer (1 elements) (OperationKind.ArrayInitializer, IsInvalid)
+    IConversionExpression (ConversionKind.Invalid, Implicit) (OperationKind.ConversionExpression, Type: System.Int32, IsInvalid)
+      IInvalidExpression (OperationKind.InvalidExpression, Type: ?, IsInvalid)
+        IArrayInitializer (2 elements) (OperationKind.ArrayInitializer, IsInvalid)
+          IConversionExpression (ConversionKind.Cast, Implicit) (OperationKind.ConversionExpression, Type: System.Object)
+            ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+          IConversionExpression (ConversionKind.Cast, Implicit) (OperationKind.ConversionExpression, Type: System.Object)
+            ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (2 elements) (OperationKind.ArrayInitializer)
+    ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+    ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+";
+            VerifyOperationTreeForTest<InitializerExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17598, "https://github.com/dotnet/roslyn/issues/17598")]
+        public void InvalidArrayCreation()
+        {
+            string source = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var x = /*<bind>*/new X[Program] { { 1 } }/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: X) (OperationKind.ArrayCreationExpression, Type: X[], IsInvalid)
+  IInvalidExpression (OperationKind.InvalidExpression, Type: Program, IsInvalid)
```

Comment:
>IInvalidExpression (OperationKind.InvalidExpression, Type: Program, IsInvalid) [](start = 2, length = 78)

Consider changing the test helper to print the name of the property, through with the sub nodes are accessed. Not in this PR though.